test/e2e/prioritize.go

```diff
@@ -0,0 +1,222 @@
+		context := initTestContext()
+		defer cleanupTestContext(context)
+
+		nodeNames := getAllWorkerNodes(context)
+		var preferredSchedulingTermSlice []v1.PreferredSchedulingTerm
+		nodeSelectorRequirement := v1.NodeSelectorRequirement{Key: kubeletapi.LabelHostname, Operator: v1.NodeSelectorOpIn, Values: []string{nodeNames[0]}}
+		nodeSelectorTerm := v1.NodeSelectorTerm{MatchExpressions: []v1.NodeSelectorRequirement{nodeSelectorRequirement}}
+		schedulingTerm := v1.PreferredSchedulingTerm{Weight: 100, Preference: nodeSelectorTerm}
+		preferredSchedulingTermSlice = append(preferredSchedulingTermSlice, schedulingTerm)
+
+		slot := oneCPU
+		_, rep := computeNode(context, oneCPU)
+		Expect(rep).NotTo(Equal(0))
+
+		affinity := &v1.Affinity{
+			NodeAffinity: &v1.NodeAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: preferredSchedulingTermSlice,
+			},
+		}
+
+		job := &jobSpec{
+			name: "pa-job",
+			tasks: []taskSpec{
+				{
+					img:      "nginx",
+					req:      slot,
+					min:      1,
+					rep:      1,
+					affinity: affinity,
+				},
+			},
+		}
+
+		_, pg := createJobEx(context, job)
+		err := waitPodGroupReady(context, pg)
+		Expect(err).NotTo(HaveOccurred())
+
+		pods := getPodOfPodGroup(context, pg)
+		//All pods should be scheduled in particular node
+		for _, pod := range pods {
+			Expect(pod.Spec.NodeName).To(Equal(nodeNames[0]))
+		}
+	})
+
+	It("Pod Affinity Test", func() {
+		context := initTestContext()
+		defer cleanupTestContext(context)
+
+		var preferredSchedulingTermSlice []v1.WeightedPodAffinityTerm
+		labelSelectorRequirement := metav1.LabelSelectorRequirement{Key: "test", Operator: metav1.LabelSelectorOpIn, Values: []string{"e2e"}}
+		labelSelector := &metav1.LabelSelector{MatchExpressions: []metav1.LabelSelectorRequirement{labelSelectorRequirement}}
+		podAffinityTerm := v1.PodAffinityTerm{LabelSelector: labelSelector, TopologyKey: "kubernetes.io/hostname"}
+		weightedPodAffinityTerm := v1.WeightedPodAffinityTerm{Weight: 100, PodAffinityTerm: podAffinityTerm}
+		preferredSchedulingTermSlice = append(preferredSchedulingTermSlice, weightedPodAffinityTerm)
+
+		_, rep := computeNode(context, oneCPU)
+		Expect(rep).NotTo(Equal(0))
+
+		labels := map[string]string{"test": "e2e"}
+
+		affinity := &v1.Affinity{
+			PodAffinity: &v1.PodAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: preferredSchedulingTermSlice,
+			},
+		}
+
+		job := &jobSpec{
+			name: "pa-job",
+			tasks: []taskSpec{
+				{
+					img:      "nginx",
+					req:      oneCPU,
+					min:      rep,
+					rep:      rep,
+					affinity: affinity,
+					labels:   labels,
+				},
+			},
+		}
+
+		_, pg := createJobEx(context, job)
+		err := waitPodGroupReady(context, pg)
+		Expect(err).NotTo(HaveOccurred())
+
+		pods := getPodOfPodGroup(context, pg)
+		// All Pods Should be Scheduled in same node
+		nodeName := pods[0].Spec.NodeName
+		for _, pod := range pods {
+			Expect(pod.Spec.NodeName).To(Equal(nodeName))
```

Comment:
`rep` means all resources from all worker nodes; they can not be dispatched into the same node. But how this test passed?

Reply:
rep is not all resources from all worker nodes.
```
	for _, node := range nodes.Items {
            ........
            ........
                alloc := kbapi.NewResource(node.Status.Allocatable)
		for slot.LessEqual(alloc) {
			alloc.Sub(slot)
			res++
		}

		if res > 0 {
			return node.Name, res
		}
	}
```
It is number of workload, for particular amount of requested resource that can be scheduled in that node.  If `res` value is greater than zero, then it returns the value for that particular worker node, it does not traverse through all the worker nodes to calculate `res`.  If for that particular workload's requested resource cannot be scheduled then `res` value will be zero, so it checks next node, if `res` is not zero, it does not check other nodes, it will return those values

Reply:
yes, your're right; so that's ok to me :)

Reply:
btw, it seems not stable for this case; would you help to check the CI result?

Reply:
Yeah checking CI failures, will fix it.